openlcs/packages/mixins.py

```diff
@@ -124,12 +124,22 @@ def build_release_node(self, release_component):
         """
         # Create release node
         release_ctype = ContentType.objects.get_for_model(Release)
-        release_node, _ = ProductTreeNode.objects.get_or_create(
-            name=self.release.name,
-            content_type=release_ctype,
-            object_id=self.release.id,
-            parent=None,
-        )
+        try:
+            release_node, _ = ProductTreeNode.objects.get_or_create(
+                name=self.release.name,
+                content_type=release_ctype,
+                object_id=self.release.id,
+                parent=None,
+            )
+        except IntegrityError:
+            # if release object name been changed, get_or_create
```

Comment:
I'm not sure I understand the `IntegrityError ` context here.

According to the official doc, `get_or_create` is stomic, without db level constraints, concurrent calls may result in multiple records created. https://docs.djangoproject.com/en/4.1/ref/models/querysets/#get-or-create .

We don't have any constraint on `Release.name` at all, but if I understand correctly, it should not be a random string, but instead should be concatenated by `Product.name`+`-`+`version`, product + version has unique constaints, thus name should also be unique. In fact I don't think it makes much sense to maintain a release.name field at all.

With that said, if `release name` is changed, it will be a different release.

Reply:
The OLCS-344 problem happened because release object's info been changed by `django admin` page.In this situation, due to db  constraint about `object_id` and `parent`. `get_or_create` will raise a IntegrityError because args pass to them will try to create a new root node since Release.name change.It will cause multiple root node.I write this code to adapt this situation.
If a Release object been imported with some package.It already has a root node.Then if it name been changed and import other package will meet this problem.Because we actually identity Release by its pk id.

Reply:
latest changes LGTM.

Thanks for the effort. Feel free to merge after your verification.

Reply:
Ok. Thanks for your review